grpc/errors.go

```diff
@@ -110,7 +111,17 @@ func unwrapError(err error, md metadata.MD) error {
 			)
 		}
 
-		unmarshalErr := json.Unmarshal([]byte(subErrorsVal[0]), &outErr.SubErrors)
+		utf8SubErrors, unquoteErr := strconv.Unquote(subErrorsVal[0])
```

Comment:
This variable name seems to assert that the result is utf8 but it's not clear to me where that is guaranteed. Strings in Go are [arbitrary byte sequences](https://go.dev/blog/strings).

Reply:
Good call, done